pandapipes/component_models/abstract_models/branch_models.py

```diff
@@ -130,26 +130,23 @@ def calculate_derivatives_thermal(cls, net, branch_pit, node_pit, idx_lookups, o
 
         transient = get_net_option(net, "transient")
 
-        tvor = branch_pit[:, T_OUT_OLD]
+        tvor = branch_component_pit[:, T_OUT_OLD]
```

Comment:
Can u explain this change? I do not quite understand why we can just change branch_pit to branch_component_pit

Reply:
branch_component_pit is related to the specific component (e.g. pipe vs. valve), while branch_pit contains entries for all components. We only want to do the calculation for the single component here. A global formulation would be nice to have, as it is more performant, but has not been implemented yet.